### ISSUE TITLE
Broken thumbnails due to http scheme issue with AWS load balancer setup

### ISSUE THREAD

**Issue by GregVes** [directus/api#1949]
Hello Directus community,

I installed a Directus Helm chart on an EKS cluster's EC2 instance and it is served by an AWS ELB that uses an SSL certificate from AWS ACM.

It works very well for most of the features (I can login, plug the api to my custom db, create collections and items etc).

Nevertheless, when reaching the file library page to fetch the thumbnails, the admin app is calling the API with "http" scheme which raises the mixed content error. Hence the broken thumbnails.

Instead, when clicking on a broken thumbnail to access the editing page, the app makes a call to get the full image using the "https" scheme which makes the request successful.

To sum up, here are the two calls made by the admin app: 

On file library route: **http:**//MY_HOST_NAME/_/assets/ASSET_ID**?key=directus-medium-crop** (FAILING)

On editing page route: **https**://MY_HOST_NAME/_/assets/ASSET_ID (SUCCESS)

I tried to add the following to my config/my-config.php, but it does not have any effect

    'HTTP' => [
        'forceHttps' => true
    ]

Could it be an issue with my load balancer? Or Something related to the query string?

**Comment by GregVes** [comment]
Problem solved, but probably not in the canonical way.

The issue seemed to be at the API level.

The files object returned when accessing the file library route have their thumbnails's "url" (used by the frontend to set the src tags) property using the http protocol (I did not went deeper than the endpoint/Files.php because of lack of time).

To fix this, I replaced each "url" value by the relative url value, so it stripped the scheme part.

It is far from being a clean solution, I hope to have some more time to find the root of the problem.
